examples/bpi-monolith-test-suite/bpi/components/storage/agreement.ts

```diff
@@ -0,0 +1,75 @@
+import { Order } from "../../../domain-objects/order";
+
+export class Agreement {
+    productIds: string[] = [];
+    orders: Order[] = [];
+    proofs: string[] = [];
+    signature: boolean = false;
+
+    addOrder(order: Order): boolean {
+        if (!this.isOrder(order)) {
+            return false;
+        }
+
+        if (!this.orderPriceIsGreater(order)) {
```

Comment:
same comment about condition requirements ... one call with a selector for the constraint for the type of domain-object and version